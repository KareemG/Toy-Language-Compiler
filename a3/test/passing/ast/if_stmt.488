{
    var a : integer
    var b, c : integer

    read a
    read b
    read c

    if a = b then
        write "a = b"

    if a + b = c then
    {
        write "valid"
    }
    else
        write "invalid"
    
    if 5 + 7 > 11 then
        if 9 - 3 < 7 then
            if a + b = c then
                write "valid"
            else
                write "valid"
        else
            write "invalid"
    else
        write "invalid"
}
