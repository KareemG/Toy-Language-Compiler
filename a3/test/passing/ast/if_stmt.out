 { 
 var a : integer
 var b, c : integer
 read a
 read b
 read c
 if (a) = (b) then
 write "a = b"end
 if ((a) + (b)) = (c) then
 { 
 write "valid"
 } else
 write "invalid"end
 if ((5) + (7)) > (11) then
 if ((9) - (3)) < (7) then
 if ((a) + (b)) = (c) then
 write "valid" else
 write "valid"end
 else
 write "invalid"end
 else
 write "invalid"end
 }
